cli/xml2rfc/writers/base.py

```diff
@@ -2216,3 +2216,39 @@ def remove(self, p, e):
                 p.text = (p.text or '') + e.tail
         p.remove(e)
 
+    @staticmethod
+    def split_pn(pn):
+        """Split a pn into meaningful parts
+
+        Returns a tuple (element_type, section_number, paragraph_number).
+        If there is no paragraph number, it will be None.
+        """
+        parts = pn.split('-', 2)
+        elt_type = parts[0]
+        sect_num = parts[1]
+        paragraph_num = parts[2] if len(parts) > 2 else None
+
+        # see if section num needs special handling
+        components = sect_num.split('.', 1)
+        if components[0] in ['appendix', 'boilerplate', 'note', 'toc']:
+            sect_num = components[1]  # these always have a second component
+        return elt_type, sect_num, paragraph_num
+
+    @staticmethod
+    def level_of_section_num(num):
+        """Determine hierarchy level of a section number
+
+        Top level items have level 1. N.B., num is a string.
+        """
+        if num[-1] == '.':
+            num = num[:-1]  # chop off trailing '.'
+        return len(num.split('.'))
```

Comment:
I think it's clearer with the two-line format. (This is a case where I really wish `len()` weren't a function because the mixing of prefix and postfix makes the one-liner confusing to parse.)